eclair-core/src/main/scala/fr/acinq/eclair/router/Router.scala

```diff
@@ -424,6 +400,106 @@ class Router(nodeParams: NodeParams, watcher: ActorRef) extends FSM[State, Data]
         .map(r => sender ! RouteResponse(r, ignoreNodes, ignoreChannels))
         .recover { case t => sender ! Status.Failure(t) }
       stay
+
+    case Event(SendChannelQuery(remoteNodeId, remote), _) =>
+      // ask for everything
+      val query = QueryChannelRange(nodeParams.chainHash, firstBlockNum = 0, numberOfBlocks = Int.MaxValue)
+      log.info("querying {} from {}", query, remoteNodeId)
+      remote ! query
+
+      // we also set a pass-all filter for now (we can update it later)
+      val filter = GossipTimestampFilter(nodeParams.chainHash, firstTimestamp = 0, timestampRange = Int.MaxValue)
+      remote ! filter
+      stay
+
+    case Event(PeerRoutingMessage(remoteNodeId, routingMessage: HasChainHash), d) if routingMessage.chainHash != nodeParams.chainHash =>
```

Comment:
This needs to be updated per #638, now that we have the source `nodeId` it should be put in an mdc.

Reply:
Maybe also add a "warning! order matters" comment because of how we handle the `chain_hash` check